src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -735,6 +735,156 @@ RppStatus rppt_ricap_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** pixelate ********************/
+
+RppStatus rppt_pixelate_host(RppPtr_t srcPtr,
+                             RpptDescPtr srcDescPtr,
+                             RppPtr_t dstPtr,
+                             RpptDescPtr dstDescPtr,
+                             Rpp32f pixelationPercentage,
+                             RpptROIPtr roiTensorPtrSrc,
+                             RpptRoiType roiType,
+                             rppHandle_t rppHandle)
+{
+    RppLayoutParams srcLayoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+    RpptDescPtr interDescPtr = srcDescPtr;
+    RpptImagePatchPtr internalDstImgSizes = static_cast<RpptImagePatch *> (calloc(dstDescPtr->n, sizeof(RpptImagePatch)));
+    RpptROI *internalRoiTensorPtrSrc = static_cast<RpptROI *>( calloc(dstDescPtr->n, sizeof(RpptROI)));
+
+    for(int i = 0 ; i < dstDescPtr->n ; i++)
+    {
+        internalDstImgSizes[i].width = internalRoiTensorPtrSrc[i].xywhROI.roiWidth = (roiTensorPtrSrc[i].xywhROI.roiWidth * (100 - pixelationPercentage)) / 100;
+        internalDstImgSizes[i].height = internalRoiTensorPtrSrc[i].xywhROI.roiHeight = (roiTensorPtrSrc[i].xywhROI.roiHeight * (100 - pixelationPercentage)) / 100;
+    }
+
+    Rpp64u interBufferSize = static_cast<Rpp64u>(interDescPtr->strides.nStride) * static_cast<Rpp64u>(interDescPtr->n);
+
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::U8))
+    {
+        Rpp8u *interDstPtr = static_cast<Rpp8u *>(calloc(interBufferSize, sizeof(Rpp8u)));
```

Comment:
This will be too large a memory to allocate on runtime. What if the image size is 4K and batch size is 512? Why are we doing this?
If really needed please ask user to pass a pointer called userScratchBufferHost and in the documentation,  add the condition that it must be the same size as (srcDescPtr->n * srcDescPtr->strides.nStride * sizeof(<bitDepth>))

Reply:
Done